ouroboros-consensus-test/src/Test/Util/Serialisation/Roundtrip.hs

```diff
@@ -228,17 +228,32 @@ instance ( blockVersion ~ BlockNodeToClientVersion blk
       -- This case statement will cause a warning when we add a new top
       -- level query and hence a new QueryVersion. In that case we should
       -- support such top level `Query` constructors in this Arbitrary instance.
-      Query.TopLevelQueryDisabled ->
-        arbitraryBlockQuery queryVersion
+      Query.TopLevelQueryDisabled -> arbitraryBlockQuery queryVersion
+      Query.QueryVersion1         -> genTopLevelQuery1
+      Query.QueryVersion2         -> genTopLevelQuery2
+    where
+      mkEntry :: QueryVersion
+        -> Query blk query
+        -> Gen
+            (WithVersion (QueryVersion, blockVersion) (SomeSecond Query blk))
+      mkEntry qv q = do
+        blockV <- arbitrary
+        return (WithVersion (qv, blockV) (SomeSecond q))
+
+      genTopLevelQuery1 = let version = Query.QueryVersion1 in
```

Comment:
Hmm. The `StyleGuide.md` is not clear on this, which surprises me.

So I'll just phrase this a style _request_ :D

```
genTopLevelQuery1 =
  let version = Query.QueryVersion1
  in frequency
       [ (15, arbitraryBlockQuery version    )
       , (1 , mkEntry version GetSystemStart )
       ]
```

Reply:
Done!